From 144202164a47eb755998755743256cc2b7138b74 Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Thu, 1 Sep 2016 22:51:05 +0000
Subject: [PATCH] ZAPI-750 Send FWAPI updates with owner_uuid for tag changes
 Reviewed by: Josh Wilsdon <jwilsdon@joyent.com> Approved by: Josh Wilsdon
 <jwilsdon@joyent.com>

---
 lib/workflows/add-nics.js    |  4 ++--
 lib/workflows/job-common.js  | 16 +++++++++-------
 lib/workflows/provision.js   |  2 +-
 lib/workflows/remove-nics.js |  4 ++--
 lib/workflows/update.js      |  4 ++--
 5 files changed, 16 insertions(+), 14 deletions(-)

diff --git a/lib/workflows/add-nics.js b/lib/workflows/add-nics.js
index ff4a163..8c3d8ac 100644
--- a/lib/workflows/add-nics.js
+++ b/lib/workflows/add-nics.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2014, Joyent, Inc.
+ * Copyright 2016, Joyent, Inc.
  */
 
 /*
@@ -21,7 +21,7 @@
 
 var async;  // stub to keep jsl happy
 var common = require('./job-common');
-var VERSION = '7.0.10';
+var VERSION = '7.0.11';
 
 
 /*
diff --git a/lib/workflows/job-common.js b/lib/workflows/job-common.js
index 75d75cd..45a91ee 100644
--- a/lib/workflows/job-common.js
+++ b/lib/workflows/job-common.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright 2016, Joyent, Inc.
  */
 
 /*
@@ -1016,21 +1016,23 @@ function updateFwapi(job, cb) {
     job.log.info({ jobParams: jobParams, update: update }, 'update params');
 
     if (Object.keys(update).length === 0 && job.params.task !== 'destroy') {
-        return cb(null, 'No properties affecting FWAPI found: not updating');
+        cb(null, 'No properties affecting FWAPI found: not updating');
+        return;
     }
 
-    update.owner_uuid = jobParams.owner_uuid;
-    update.server_uuid = jobParams.server_uuid;
+    update.owner_uuid = jobParams.owner_uuid || job.params.owner_uuid;
+    update.server_uuid = jobParams.server_uuid || job.params.server_uuid;
     update.type = type;
     update.uuid = jobParams.uuid || jobParams.vm_uuid || job.params.vm_uuid;
 
-    return fwapi.createUpdate(update, function (err, obj) {
+    fwapi.createUpdate(update, function (err, obj) {
         if (err) {
             job.log.warn(err, 'Error sending update to FWAPI');
-            return cb(null, 'Error updating FWAPI');
+            cb(null, 'Error updating FWAPI');
+            return;
         }
 
-        return cb(null, 'Updated FWAPI with update UUID: ' + obj.update_uuid);
+        cb(null, 'Updated FWAPI with update UUID: ' + obj.update_uuid);
     });
 }
 
diff --git a/lib/workflows/provision.js b/lib/workflows/provision.js
index 4de4ece..8a1e27b 100644
--- a/lib/workflows/provision.js
+++ b/lib/workflows/provision.js
@@ -19,7 +19,7 @@ var common = require('./job-common');
 var childProcess = require('child_process');
 var wfapiUrl;
 
-var VERSION = '7.3.0';
+var VERSION = '7.3.1';
 
 
 /*
diff --git a/lib/workflows/remove-nics.js b/lib/workflows/remove-nics.js
index cac08a9..032df1c 100644
--- a/lib/workflows/remove-nics.js
+++ b/lib/workflows/remove-nics.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright 2016, Joyent, Inc.
  */
 
 /*
@@ -15,7 +15,7 @@
 var common = require('./job-common');
 var async = require('async');
 
-var VERSION = '7.0.5';
+var VERSION = '7.0.6';
 
 
 /*
diff --git a/lib/workflows/update.js b/lib/workflows/update.js
index 55d8a40..fd44679 100644
--- a/lib/workflows/update.js
+++ b/lib/workflows/update.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2014, Joyent, Inc.
+ * Copyright 2016, Joyent, Inc.
  */
 
 /*
@@ -15,7 +15,7 @@
 var restify = require('restify');
 var common = require('./job-common');
 
-var VERSION = '7.1.8';
+var VERSION = '7.1.9';
 
 /*
  * Sets up a CNAPI VM action request. Take a look at common.zoneAction. Here you
-- 
2.21.0

